packages/categorize/configure/src/design/choices/index.jsx

```diff
@@ -110,21 +115,23 @@ export class Choices extends React.Component {
       defaultImageMaxWidth,
       defaultImageMaxHeight,
     } = this.props;
-    const { errors, allowMultiplePlacementsEnabled, lockChoiceOrder } = model;
+    const { errors, allowMultiplePlacementsEnabled, lockChoiceOrder, maxAnswerChoices } = model;
     const { choicesError, choicesErrors } = errors || {};
-    const { maxChoices, maxImageWidth = {}, maxImageHeight = {} } = configuration || {};
-
+    const { maxImageWidth = {}, maxImageHeight = {} } = configuration || {};
     const choiceHolderStyle = {
       gridTemplateColumns: `repeat(${model.categoriesPerRow}, 1fr)`,
     };
+    const addChoiceButtonTooltip =
+        maxAnswerChoices && choices?.length >= maxAnswerChoices ? `Only ${maxAnswerChoices} allowed maximum` : '';
 
     return (
       <div className={classNames(classes.choices, className)}>
         <Header
           label="Choices"
           buttonLabel="ADD A CHOICE"
           onAdd={this.addChoice}
-          buttonDisabled={maxChoices && choices && maxChoices === choices.length}
+          buttonDisabled={maxAnswerChoices && choices && maxAnswerChoices === choices?.length}
```

Comment:
You’re right - if maxAnswerChoices is set to a lower number than the number of choices, the button won't be disabled initially. 
To address this, I can adjust the logic so that the button is disabled when the number of choices is greater than or equal to maxAnswerChoices. So, in this particular case the button will be disabled at first.
We should note that the button will be re-enabled only when the user will deleted choices and the number will fall below the maxAnswerChoices.